lib/srv/usermgmt_linux.go

```diff
@@ -72,7 +72,15 @@ func newHostSudoersBackend(uuid string) (HostSudoersBackend, error) {
 
 // Lookup implements host user information lookup
 func (*HostUsersProvisioningBackend) Lookup(username string) (*user.User, error) {
-	return user.Lookup(username)
+	user, err := user.Lookup(username)
+	if err != nil {
+		if strings.Contains(err.Error(), "no such file or directory") {
```

Comment:
I tried to detect this error a couple of different ways, but a string contains was the only one that ended up working. For some reason `users.Lookup()` eats the original error context because it formats with `%s` instead of `%w`.

Reply:
Should we first check if error is [UnknownUserError](https://pkg.go.dev/os/user#UnknownUserError) so that we check for a known error message first before doing string contains?

Reply:
Is this sort of what you had in mind @flyinghermit? Since the handling for `UnknownUserError` is the same as `err != nil` I just added a negative check before running the string contains. The error itself in this scenario doesn't actually contain `UnkownUserError` so I don't think there's a better way to avoid unnecessary containment checks 🤔

Reply:
Got it, Thanks for clarifying.

My suggestion originated more for readability. Reading the godoc for `user.Lookup` it is not apparent that the `no such file or directory` error would be return. So explicitly including `UnknownUserError` check in the error block makes it clear that there may be error besides `UnknownUserError`.